tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -265,13 +265,34 @@ JNIEXPORT jobject JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1prepare(JNI
                                                                              jbyteArray query_j) {
 	auto conn_ref = (Connection *)env->GetDirectBufferAddress(conn_ref_buf);
 	if (!conn_ref) {
-		env->ThrowNew(J_SQLException, "Invalid connection");
+		env->ThrowNew(env->FindClass("java/sql/SQLException"), "Invalid connection");
```

Comment:
With commit ae0f79afbe9992dd9020291000a1c07c52212051 there `env->FindClass` calls can be avoided. Maybe you change this accordingly to something like: `env->ThrowNew(J_SQLException, "reason ...");`